### ISSUE TITLE
Cannot merge sp3 files with sp3merge properly

### ISSUE THREAD

**Issue by different91988** [GeoscienceAustralia/gnssanalysis#28]
Two SP3 files download from IGS AC cannot be merged properly. 
The file urls are followings:
ftp://igs.gnsswhu.cn/pub/gps/products/2286/WUM0MGXFIN_20233020000_01D_05M_ORB.SP3.gz
ftp://igs.gnsswhu.cn/pub/gps/products/2286/USN0OPSULT_20233040000_02D_15M_ORB.SP3.gz
Same problem occured when try to merge other IGS MGEX SP3 files. The exception throws when try to merge two dataframes read from sp3 files in Line 469 of file sp3.py eg. merged_sp3 = _pd.concat(sp3_dfs)

Traceback (most recent call last):
  File "D:\Program Files\JetBrains\PyCharm 2024.1.3\plugins\python\helpers\pydev\pydevd.py", line 1546, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "D:\Program Files\JetBrains\PyCharm 2024.1.3\plugins\python\helpers\pydev\_pydev_imps\_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "D:\pythonProj\gnssanalysis\tests\test_diff.py", line 8, in <module>
    merged_df = _gn_io.sp3.sp3merge(["D:\\MATLAB\\compareSp3\\code\\WUM0MGXFIN_20232020000_01D_05M_ORB.SP3",
  File "D:\pythonProj\gnssanalysis\gnssanalysis\gn_io\sp3.py", line 469, in sp3merge
    merged_sp3 = _pd.concat(sp3_dfs)
  File "D:\Anaconda\envs\myenv\lib\site-packages\pandas\core\reshape\concat.py", line 385, in concat
    return op.get_result()
  File "D:\Anaconda\envs\myenv\lib\site-packages\pandas\core\reshape\concat.py", line 623, in get_result
    return cons(new_data).__finalize__(self, method="concat")
  File "D:\Anaconda\envs\myenv\lib\site-packages\pandas\core\generic.py", line 5963, in __finalize__
    check_attrs = all(objs.attrs == attrs for objs in other.objs[1:])
  File "D:\Anaconda\envs\myenv\lib\site-packages\pandas\core\generic.py", line 5963, in <genexpr>
    check_attrs = all(objs.attrs == attrs for objs in other.objs[1:])
  File "D:\Anaconda\envs\myenv\lib\site-packages\pandas\core\ops\common.py", line 81, in new_method
    return method(self, other)
  File "D:\Anaconda\envs\myenv\lib\site-packages\pandas\core\arraylike.py", line 40, in __eq__
    return self._cmp_method(other, operator.eq)
  File "D:\Anaconda\envs\myenv\lib\site-packages\pandas\core\series.py", line 6090, in _cmp_method
    raise ValueError("Can only compare identically-labeled Series objects")
ValueError: Can only compare identically-labeled Series objects

**Comment by ronaldmaj** [comment]
Hi [different91988](https://github.com/different91988),

Thanks for using `gnssanalysis` and bringing this issue to our attention. I will be having a closer look at this but it seems like there may be an issue with the way we merge dataframes together - the header information needs to be the same for it to work (as implied by the final error message: `ValueError: Can only compare identically-labeled Series objects`).
This may be caused by the fact that sp3 files from two different ACs are being merged. It may be a temporary fix if you change the header information to mimic two files from the same AC. 
If this is the problem, I'll work on putting a permanent fix to this soon

**Comment by different91988** [comment]
Hi @ronaldmaj ,

Thanks for your timely reply, I have to tell you that this issue also occured when I try to merge two SP3 file from same IGS AC eg. WHU or GFZ. I have tested with following two pairs of files, but same exception reports.
The tested file urls are:
ftp://igs.gnsswhu.cn/pub/gps/products/2286/WUM0MGXFIN_20233020000_01D_05M_ORB.SP3.gz 
AND
ftp://igs.gnsswhu.cn/pub/gps/products/2286/WUM0MGXFIN_20232030000_01D_05M_ORB.SP3.gz
OR
ftp://igs.gnsswhu.cn/pub/gps/products/2286/GFZ0OPSRAP_20233020000_01D_05M_ORB.SP3.gz
AND
ftp://igs.gnsswhu.cn/pub/gps/products/2286/GFZ0OPSRAP_20233030000_01D_05M_ORB.SP3.gz

**Comment by ronaldmaj** [comment]
Hi [different91988](https://github.com/different91988),

Apologies for taking a while to get back to this. I've now had a good look at this and downloaded GFZ / WHU files to test against.

I think I have found the issue and it was with how our code was set up. When the two sp3 files are read into DataFrames, their `df.attrs` are different.
This means that the `pd.concat` function used within `sp3merge` fails to properly concatenate the DataFrames and causes an error. 

Please see the following branch to see the resolution I will be putting through for this:
https://github.com/GeoscienceAustralia/gnssanalysis/tree/NPI-3429-fix-sp3merge-diff-attrs-issue

When testing this I have used two files from the same Analysis Centre (GFZ):
```sh 
sp3merge -s GFZ0OPSRAP_20233020000_01D_05M_ORB.SP3 -s GFZ0OPSRAP_20233030000_01D_05M_ORB.SP3 -o aux/ex.sp3
```
and also with two files from different Analysis Centres (GFZ / WHU):
```
sp3merge -s GFZ0OPSRAP_20233020000_01D_05M_ORB.SP3 -s WUM0MGXFIN_20233030000_01D_05M_ORB.SP3 -o aux/ex2.sp3
```
I've attached the resultant files to this message:
[example-sp3merge-GFZ-WHU.zip](https://github.com/user-attachments/files/16423309/example-sp3merge-GFZ-WHU.zip)


Thank you for bringing this to our attention, I really appreciate it. It's an issue that's been overlooked for a while.

Please let me know if you have any other questions around this or any other part of the `gnssanalysis` library

**Comment by ronaldmaj** [comment]
Hi [different91988](https://github.com/different91988),

Just to add to my comment above, we have now released Version 0.0.55 which contains the fix for the `sp3merge` utility.

You should be able to `pip install` for the latest version
```
pip install gnssanalysis -U
```
and then run the command as before
```
sp3merge -s file1.sp3 -s file2.sp3 -o output_dir/merged_file.sp3
```
in order merge your files together. I have tested it with `GFZ` and `WHU` files as in your original file and this now works fine. 
There are some assumptions used for the output header when the files to be merged have different headers, so please read the underlying code to determine how this will come out: https://github.com/GeoscienceAustralia/gnssanalysis/blob/main/gnssanalysis/gn_io/sp3.py#L718

Any more questions, please do not hesitate to reach out
